Lib/test/test_list.py

```diff
@@ -365,5 +365,19 @@ def lappend(l, x, y):
         rc, _, _ = assert_python_ok("-c", code)
         self.assertEqual(rc, 0)
 
+    def test_list_overwrite_local(self):
+        """Test that overwriting the last reference to the
+           iterable doesn't prematurely free the iterable"""
+
+        def foo(x):
+            r = 0
```

Comment:
I'd add this to make sure the test is testing what the comment is saying.
`assert(sys.getrefcount(x) == 1)`